### PR TITLE
Refactor PublicKeyCredentialCreationOptions

### PR DESCRIPTION
**Pull request by SaintPatrck**
## 🎟️ Tracking

N/A

## 📔 Objective

Rename PublicKeyCredentialCreationOptions to PasskeyAttestationOptions. Relocated it to the data model package since it is not a network model.

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed
  issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: app/src/test/java/com/x8bit/bitwarden/ui/vault/feature/addedit/util/PublicKeyCredentialCreationOptionsTestHelpers.kt [hunk 1]
```diff
@@ -1,37 +1,37 @@
...
-import com.x8bit.bitwarden.data.autofill.fido2.datasource.network.model.PublicKeyCredentialCreationOptions
+import com.x8bit.bitwarden.data.autofill.fido2.model.PasskeyAttestationOptions
 
 /**
- * Returns a mock FIDO 2 [PublicKeyCredentialCreationOptions] object to simulate a credential
+ * Returns a mock FIDO 2 [PasskeyAttestationOptions] object to simulate a credential
  * creation request.
  */
 @Suppress("MaxLineLength")
 fun createMockPublicKeyCredentialCreationOptions(
```
[the remark this answers is not in the document]
Comment (SaintPatrck) [reply app/src/test/java/com/x8bit/bitwarden/ui/vault/feature/addedit/util/PublicKeyCredentialCreationOptionsTestHelpers.kt:10]: Yes. Thank you!

File: app/src/main/java/com/x8bit/bitwarden/data/autofill/fido2/manager/Fido2CredentialManager.kt [hunk 2]
```diff
@@ -27,7 +27,7 @@ interface Fido2CredentialManager {
     /**
      * Attempt to extract FIDO 2 passkey creation options from the system [requestJson], or null.
```
Comment (shannon-livefront) [inline app/src/main/java/com/x8bit/bitwarden/data/autofill/fido2/manager/Fido2CredentialManager.kt:28]:
```suggestion
     * Attempt to extract FIDO 2 passkey attestation options from the system [requestJson], or null.
```

File: app/src/main/java/com/x8bit/bitwarden/ui/vault/feature/addedit/VaultAddEditViewModel.kt [hunk 3]
```diff
@@ -119,7 +119,7 @@ class VaultAddEditViewModel @Inject constructor(
             val fido2CreationOptions = fido2CreationRequest
```
Comment (shannon-livefront) [inline app/src/main/java/com/x8bit/bitwarden/ui/vault/feature/addedit/VaultAddEditViewModel.kt:119]:
```suggestion
            val fido2AttestationOptions = fido2CreationRequest
```

File: app/src/main/java/com/x8bit/bitwarden/ui/vault/feature/itemlisting/VaultItemListingViewModel.kt [hunk 4]
```diff
@@ -448,7 +448,7 @@ class VaultItemListingViewModel @Inject constructor(
         cipherView: CipherView,
     ) {
         val createOptions = fido2CredentialManager
```
Comment (shannon-livefront) [inline app/src/main/java/com/x8bit/bitwarden/ui/vault/feature/itemlisting/VaultItemListingViewModel.kt:450]:
```suggestion
        val attestationOptions = fido2CredentialManager
```

File: app/src/test/java/com/x8bit/bitwarden/data/autofill/fido2/manager/Fido2CredentialManagerTest.kt [hunk 5]
```diff
@@ -269,8 +269,8 @@ class Fido2CredentialManagerTest {
     fun `getPasskeyCreateOptionsOrNull should return passkey options when deserialized`() =
```
Comment (shannon-livefront) [inline app/src/test/java/com/x8bit/bitwarden/data/autofill/fido2/manager/Fido2CredentialManagerTest.kt:269]:
```suggestion
    fun `getPasskeyAttestationOptionsOrNull should return passkey options when deserialized`() =
```

File: app/src/test/java/com/x8bit/bitwarden/ui/vault/feature/addedit/VaultAddEditViewModelTest.kt [hunk 6]
```diff
@@ -732,7 +732,7 @@ class VaultAddEditViewModelTest : BaseViewModelTest() {
             val mockCreateResult = Fido2RegisterCredentialResult.Success(
                 registrationResponse = "mockRegistrationResponse",
             )
-            val mockCreateOptions = createMockPublicKeyCredentialCreationOptions(
+            val mockCreateOptions = createMockPasskeyAttestationOptions(
```
Comment (shannon-livefront) [inline app/src/test/java/com/x8bit/bitwarden/ui/vault/feature/addedit/VaultAddEditViewModelTest.kt:735]:
```suggestion
            val mockAttestationOptions = createMockPasskeyAttestationOptions(
```

File: app/src/test/java/com/x8bit/bitwarden/ui/vault/feature/addedit/VaultAddEditViewModelTest.kt [hunk 7]
```diff
@@ -811,7 +811,7 @@ class VaultAddEditViewModelTest : BaseViewModelTest() {
                 ),
             )
             val mockCreateResult = Fido2RegisterCredentialResult.Success("mockResponse")
-            val mockCreateOptions = createMockPublicKeyCredentialCreationOptions(
+            val mockCreateOptions = createMockPasskeyAttestationOptions(
```
Comment (shannon-livefront) [inline app/src/test/java/com/x8bit/bitwarden/ui/vault/feature/addedit/VaultAddEditViewModelTest.kt:814]:
```suggestion
            val mockAttestationOptions = createMockPasskeyAttestationOptions(
```

**Review by shannon-livefront — APPROVED** [review]

**Review by shannon-livefront — APPROVED** [review]

**Pull request merged**